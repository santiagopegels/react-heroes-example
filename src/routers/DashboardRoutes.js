import React from 'react'
import { Navbar } from '../components/ui/NavBar'
import { Switch, Route, Redirect } from 'react-router-dom'
import { MarvelScreen } from '../components/marvel/MarvelScreen'
import { HeroesScreen } from '../components/heroes/HeroesScreen'
import { DcScreen } from '../components/dc/DcScreen'

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar />

            <div className="container mt-3">
                <Switch>
                    <Route exact path="/marvel" component={MarvelScreen}/>
                    <Route exact path="/dc" component={DcScreen}/>
                    <Route exact path="/hero/:heroId" component={HeroesScreen}/>

                    <Redirect to="/marvel" />
                </Switch>
            </div>
        </>
    )
}
