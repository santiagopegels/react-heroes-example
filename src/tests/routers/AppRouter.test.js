import { mount } from "enzyme"
import { AppRouter } from "../../routers/AppRouter"
import { AuthContext } from "../../auth/AuthContext"

describe('Test on <AppRouter/>', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    }

    test('should show login view if the user is not authenticated', () => {

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        )

        expect(wrapper).toMatchSnapshot()

    })

    test(`should marvel's component if the user is authenticated`, () => {
        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'Test'
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        )

        expect(wrapper.find('.navbar').exists()).toBe(true)

    })


})
