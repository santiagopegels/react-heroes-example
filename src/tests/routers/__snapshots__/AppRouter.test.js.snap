// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <AppRouter/> should show login view if the user is not authenticated 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      LoginScreen
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
